### PR TITLE
Expose validate error

### PR DESCRIPTION
**Pull request by SimonRichardson**
The error during validation was being trapped, we need to ensure that
the error bubbles up so that we can show the issue to the user.

### CODE REVIEW

File: manifest.go [hunk 1]
```diff
@@ -30,7 +29,7 @@ func NewManifest() *Manifest {
 func (m *Manifest) Validate() error {
 	for _, b := range m.Bases {
 		if err := b.Validate(); err != nil {
-			return fmt.Errorf("invalid base: empty file")
+			return errors.Annotate(err, "invalid base")
```
Comment (manadart) [inline manifest.go:32]: "validating base"

File: manifest_test.go [hunk 2]
```diff
@@ -44,3 +44,21 @@ bases:
...
 	}})
 }
+
+func (s *manifestSuite) TestReadValidateManifest(c *gc.C) {
+	_, err := ReadManifest(strings.NewReader(`
+bases:
+  - name: ""
+    channel: "18.04"
+`))
+	c.Assert(err, gc.ErrorMatches, "manifest: name must be specified not valid")
```
Comment (manadart) [inline manifest_test.go:54]: Ultimate emission should be "base without name not valid".

**Review by manadart — CHANGES_REQUESTED** [review]
Wording suggestions.

**Review by manadart — APPROVED** [review]
Much better.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**